File: app/src/main/java/com/app/customers/data/model/MerchantCustomer.java
package com.app.customers.data.model;

import android.os.Parcel;
import android.os.Parcelable;
import android.os.Parcelable.Creator;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class MerchantCustomer implements Parcelable
{

    @SerializedName("id")
    @Expose
    private int id;
    public final static Parcelable.Creator<MerchantCustomer> CREATOR = new Creator<MerchantCustomer>() {


        @SuppressWarnings({
                "unchecked"
        })
        public MerchantCustomer createFromParcel(Parcel in) {
            return new MerchantCustomer(in);
        }

        public MerchantCustomer[] newArray(int size) {
            return (new MerchantCustomer[size]);
        }

    }
            ;

    protected MerchantCustomer(Parcel in) {
        this.id = ((int) in.readValue((int.class.getClassLoader())));
    }

    /**
     * No args constructor for use in serialization
     *
     */
    public MerchantCustomer() {
    }

    /**
     *
     * @param id
     */
    public MerchantCustomer(int id) {
        super();
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(id);
    }

    public int describeContents() {
        return 0;
    }

}

File: app/src/main/java/com/app/customers/data/model/MerchantCustomer.kt
package com.app.customers.data.model

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class MerchantCustomer : Parcelable {

    @SerializedName("id")
    @Expose
    var id: Int = 0

    protected constructor(`in`: Parcel) {
        this.id = `in`.readValue(Int::class.javaPrimitiveType!!.classLoader) as Int
    }

    /**
     * No args constructor for use in serialization
     *
     */
    constructor() {}

    /**
     *
     * @param id
     */
    constructor(id: Int) : super() {
        this.id = id
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(id)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<MerchantCustomer> = object : Creator<MerchantCustomer> {


            override fun createFromParcel(`in`: Parcel): MerchantCustomer {
                return MerchantCustomer(`in`)
            }

            override fun newArray(size: Int): Array<MerchantCustomer?> {
                return arrayOfNulls(size)
            }

        }
    }

}